python/watch_hook.py

```diff
@@ -213,6 +213,10 @@ def load_yaml(self, yaml_stream):
                         # At the moment, we stuff the resolver's datacenter into the association
                         # ID for this watch. The ResourceFetcher relies on that.
 
+                        assert resolver.datacenter
```

Comment:
Same comment as earlier, what happens if these assertions fail, is this a bug if they fail or should we be handling these gracefully?